applications/trilinos_application/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -567,6 +567,25 @@ class TrilinosBlockBuilderAndSolver
         if (BaseType::mDofSet.size()==0)
             KRATOS_ERROR << "No degrees of freedom!";
 
+    // If reactions are to be calculated, we check if all the dofs have reactions defined
+    // This is tobe done only in debug mode
+
+    #ifdef KRATOS_DEBUG        
+
+    if(BaseType::GetCalculateReactionsFlag())
+    {
+        int num_dofs =  static_cast<int> (BaseType::mDofSet.size());
```

Comment:
why didn't you collapse this in a single iterator loop?

Reply:
i am suggesting

        for(auto dof_iterator = BaseType::mDofSet.begin(); dof_iterator != BaseType::mDofSet.end(), ++dof_iterator)
        { 
            KRATOS_ERROR_IF_NOT(dof_iterator->HasReaction()) << "Reaction variable not set for the following : "<<std::endl
                << "Node : "<<dof_iterator->Id()<< std::endl
                << "Dof : "<<(*dof_iterator)<<std::endl<<"Not possible to calculate reactions."<<std::endl;
        }

Reply:
@RiccardoRossi  I left it like that (similar to the implementation in core) to be able to have a openmp pragma for the loop. I Wanted to ask  this anyway. 

@jcotela  I saw in other parallel / trillinos schemes (trilinos_residual_based_bossak_displacement_scheme) there are some openmp parallel loops. then can we also openmp parallelize the loop here ?

Reply:
@adityaghantasala the point is that OpenMP and error throwing are not compatible, meaning that many time you will not correctly see the error output when you throw from within a parallel region (it does work __at times__)

still in this case it is not crucial since the check is only in debug mode, so performance is not critical

Reply:
Oh I see. I made the suggested changed and pushed them.